src/components/Search/parsers/image.parser.ts

```diff
@@ -0,0 +1,67 @@
+import { UiState } from 'instantsearch.js';
+import { z } from 'zod';
+import { removeEmpty } from '~/utils/object-helpers';
+import { QS } from '~/utils/qs';
+import { InstantSearchRoutingParser, searchParamsSchema } from './base';
+
+export const ImagesSearchIndexSortBy = [
+  'images:rank.reactionCountAllTimeRank:asc',
+  'images:rank.commentCountAllTimeRank:asc',
+  'images:createdAt:desc',
+] as const;
+
+const imageSearchParamsSchema = searchParamsSchema
+  .extend({
+    index: z.literal('images'),
+    sortBy: z.enum(ImagesSearchIndexSortBy),
+    tags: z
+      .union([z.array(z.string()), z.string()])
+      .transform((val) => (Array.isArray(val) ? val : [val])),
+  })
+  .partial();
+
+export type ImageSearchParams = z.output<typeof imageSearchParamsSchema>;
+
+export const imagesInstantSearchRoutingParser: InstantSearchRoutingParser = {
+  parseURL: ({ location }) => {
+    const articleSearchIndexResult = imageSearchParamsSchema.safeParse(QS.parse(location.search));
+    const articleSearchIndexData: ImageSearchParams | Record<string, string[]> =
+      articleSearchIndexResult.success ? articleSearchIndexResult.data : {};
+
+    return { images: removeEmpty(articleSearchIndexData) };
```

Comment:
This might have been leftover from copy/pasting (blame me 😬)
```suggestion
    const imageSearchIndexResult = imageSearchParamsSchema.safeParse(QS.parse(location.search));
    const imageSearchIndexData: ImageSearchParams | Record<string, string[]> =
      imageSearchIndexResult.success ? imageSearchIndexResult.data : {};

    return { images: removeEmpty(imageSearchIndexData) };
```